src/pkg/cli/compose/context.go

```diff
@@ -114,7 +114,12 @@ func uploadTarball(ctx context.Context, provider client.Provider, project string
 		return "", fmt.Errorf("HTTP PUT failed with status code %v", resp.Status)
 	}
 
-	return http.RemoveQueryParam(res.Url), nil
+	url := http.RemoveQueryParam(res.Url)
+	const gcpPrefix = "https://storage.googleapis.com/"
+	if strings.HasPrefix(url, gcpPrefix) {
+		url = "gs://" + url[len(gcpPrefix):]
+	}
```

Comment:
@edwardrf why is this logic here?

Reply:
Because using the gs:// url, cd will use gcp SDK which is authenticated, whereas the HTTPS URL need extra authorization setup, which we have not implemented.